testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -1915,6 +1921,163 @@ fn microblock_integration_test() {
+        .unwrap();
+
+    // Make microblock with two transactions.
+    let recipient = StacksAddress::from_string(ADDR_4).unwrap();
+    let transfer_tx =
+        make_stacks_transfer_mblock_only(&spender_sk, 0, 1000, &recipient.into(), 1000);
+
+    let caller_src = "
+    (define-public (execute)
+       (ok stx-liquid-supply))
+    ";
+    let publish_tx =
+        make_contract_publish_microblock_only(&spender_sk, 1, 1000, "caller", caller_src);
+
+    let tx_1 = StacksTransaction::consensus_deserialize(&mut &transfer_tx[..]).unwrap();
+    let tx_2 = StacksTransaction::consensus_deserialize(&mut &publish_tx[..]).unwrap();
+    let vec_tx = vec![tx_1, tx_2];
+    let privk =
+        find_microblock_privkey(&conf, &stacks_block.header.microblock_pubkey_hash, 1024).unwrap();
+    let mblock = make_microblock(
+        &privk,
+        &mut chainstate,
+        &btc_regtest_controller.sortdb_ref().index_conn(),
+        consensus_hash,
+        stacks_block.clone(),
+        vec_tx,
+    );
+
+    // Store the microblock we just created, and reload the unconfirmed state to apply that microblock to the unconfirmed state.
+    assert!(chainstate
+        .preprocess_streamed_microblock(&consensus_hash, &stacks_block.block_hash(), &mblock)
+        .unwrap());
+    chainstate
+        .reload_unconfirmed_state(&btc_regtest_controller.sortdb_ref().index_conn(), tip_hash)
+        .unwrap();
+
+    // Call the contract we just defined in a microblock, but make the mempool admission check occur against anchored state.
+    let call_tx = make_contract_call(
+        &spender_sk,
+        2,
+        1000,
+        &spender_stacks_addr,
+        "caller",
+        "execute",
+        &[],
+    );
+    let client = reqwest::blocking::Client::new();
+    let query =
+        HttpRequestType::make_query_string_for_post_tx_endpoint(TransactionAnchorMode::OnChainOnly);
+    let path = format!("{}/v2/transactions{}", http_origin, query);
+    let res = client
+        .post(&path)
+        .header("Content-Type", "application/octet-stream")
+        .body(call_tx.clone())
+        .send()
+        .unwrap();
+    let err_str = res.text().unwrap();
+    // Expect to get "NoSuchContract" because the function we are attempting to call is in a
+    // contract that only exists on unconfirmed state (and we set `mempool_admission_check` to
+    // OnChainOnly).
+    assert!(err_str.contains("NoSuchContract"));
+
+    // The function `submit_tx` panics if the tx is unable to be submitted.
+    // Call the contract defined in the microblock, but make mempool admission check against unconfirmed state.
+    submit_tx(&http_origin, &call_tx, TransactionAnchorMode::OffChainOnly);
+    // Call the contract defined in the microblock. Set the transaction anchor mode to `Any`, which
+    // makes mempool admission check occur against the unconfirmed state.
+    submit_tx(&http_origin, &call_tx, TransactionAnchorMode::Any);
```

Comment:
Can you expand this test to verify that posting a transaction that will validate against the confirmed chain tip will work, even if there's no unconfirmed state available?  This would be the case where `mempool_admission_check` is not given, and there are no unconfirmed microblocks to consider.  This is to make sure that the inner call to `maybe_read_only_clarity_tx()` gets the correct chain tip, since the unconfirmed state will exist but will not be readable (since it hasn't processed any microblocks yet).